src/app/plugins/gamble.plugin.ts

```diff
@@ -0,0 +1,94 @@
+import { MessageEmbed } from 'discord.js';
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType } from '../../common/types';
+
+export default class GamblePlugin extends Plugin {
+  public commandName: string = 'gamble';
+  public name: string = 'Gamble Plugin';
+  public description: string = 'Bet your points on a coin flip';
+  public usage: string = 'gamble 100\ngamble all';
+  public override pluginAlias = [];
+  public permission: ChannelType = ChannelType.All;
+
+  public override commandPattern: RegExp = /(all|\d+)/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const userDoc = await this.container.pointService.getUserPointDoc(message.author.id);
+
+    if (userDoc.numPoints === 0) {
+      message.reply('You have no points to gamble!');
+      return;
+    }
+
+    if (args[0] === 'all') {
+      message.reply({
+        embeds: [await this._gamble(message.author.id, userDoc.numPoints, userDoc.numPoints)],
+      });
+      return;
+    }
+
+    // Try to gamble the number given by user
+    message.reply({
+      embeds: [await this._gamble(message.author.id, userDoc.numPoints, parseInt(args[0]))],
+    });
```

Comment:
Just a nit if you like to condense the code
```suggestion
    // gamble all user points if command argument supplied
    // otherwise gamble amount of points user specified
    const pointsToGamble = args[0] === 'all' ? userDoc.numPoints : parseInt(args[0])

    // Try to gamble the number given by user
    message.reply({
      embeds: [await this._gamble(message.author.id, userDoc.numPoints, pointsToGamble)],
    });
```